### DISCUSSION TITLE
[Plugin] Plugin installed cause vue error

### DISCUSSION THREAD

**Discussion by LiHowe** [vuepress/core#543] [Q&A] [answered]
I have developed a plugin by referring to the source code --> [vuepress-plugin-mermaid-next](https://www.npmjs.com/package/vuepress-plugin-mermaid-next)

It worked correctly in using `Local Plugin` way.
```TypeScript
plugins: [
    [path.resolve(__dirname, './plugins/plugin-mermaid/node/index.ts'), {}]
]
```

But, when i publish it to the `npm`, and re-reference this plugin, It cause some warning.
```TypeScript
plugins: [
    ['mermaid-next', {}]
]
```

the warnings:

```shell
WARNING in ./node_modules/vuepress-plugin-mermaid-next/lib/client/Mermaid.js 5:15-30
export 'defineComponent' (imported as 'defineComponent') was not found in 'vue' (possible exports: default)

WARNING in ./node_modules/vuepress-plugin-mermaid-next/lib/client/Mermaid.js 39:12-21
export 'onUpdated' (imported as 'onUpdated') was not found in 'vue' (possible exports: default)

WARNING in ./node_modules/vuepress-plugin-mermaid-next/lib/client/Mermaid.js 40:8-17
export 'onMounted' (imported as 'onMounted') was not found in 'vue' (possible exports: default)

WARNING in ./node_modules/vuepress-plugin-mermaid-next/lib/client/Mermaid.js 41:21-22
export 'h' (imported as 'h') was not found in 'vue' (possible exports: default)

WARNING in ./node_modules/vuepress-plugin-mermaid-next/lib/client/Mermaid.js 44:12-13
export 'h' (imported as 'h') was not found in 'vue' (possible exports: default)

webpack compiled with 5 warnings

```

so, how can i avoid these warnings or how to get my plugin worked? Thx.

**Comment by Mister-Hope** [comment, marked as answer]
We don't have times testing these types of help, but here are some suggestions, `vuepress-plugin-md-enhance` is supporting some markdown enhance features (including mermaid), and already supports v2 since last year, maybe you can check its code, and see how it's working well with latest vuepress2 with both vite and webpack.

**Reply by LiHowe** [reply]
Okey. I'll refer to this repo for modification. Thx.
